src/foam/core/stdlib.js

```diff
@@ -361,8 +361,8 @@ foam.LIB({
     function clone(o) { return o; },
     function equals(a, b) { return a === b; },
     function compare(a, b) {
-      return ( b === null || b === undefined ) ? 1 :
-          a < b ? -1 : a > b ? 1 : 0;
+      return ( b === null || b === undefined || foam.Object.isInstance(b) ||
+          foam.core.FObject.isInstance(b) ) ? 1 : a < b ? -1 : a > b ? 1 : 0;
```

Comment:
This seems really complex. Shouldn't we have something like `Number.isNumber(b) ? a > b : 1;`? Anything that comes before this in the precedence list will never hit this code path, and anything that comes after aught to return `1`, right?